mypy/checker.py

```diff
@@ -1044,6 +1045,9 @@ def check_func_item(
 
         if name == "__exit__":
             self.check__exit__return_type(defn)
+        if name == "__post_init__":
+            if dataclasses_plugin.is_processed_dataclass(defn.info):
+                dataclasses_plugin.check_post_init(self, defn, defn.info)
```

Comment:
This is unfortunate, since usually the default plugin uses the public hooks that others plugins can use. This may be fine as a workaround, but long term we should extend the plugin system to accommodate this.

Reply:
Yes, I agree. I also plan to add `attrs` plugin here as well. I will open a new issue to discuss a new hook / API to do it in a proper way 👍

Reply:
https://github.com/python/mypy/issues/15515